### ISSUE TITLE
Add quiver, vector or vector field plot

### ISSUE THREAD

**Issue by alshan** [JetBrains/lets-plot#738]
from [discussion](https://github.com/JetBrains/lets-plot/discussions/611#discussioncomment-5361403)

> -5- Function for quiver, vector or vector field plot (exist in Matplotlib and Gadfly, http://gadflyjl.org/v0.6/lib/geoms/geom_vectorfield.html).

**Comment by alshan** [comment]
See also ggplot [geom_spoke](https://ggplot2.tidyverse.org/reference/geom_spoke.html).

**Comment by alshan** [comment]
Hi @egayer, I hope you are well and fine) We've added `geom_spoke` in [v4.2.0](https://github.com/JetBrains/lets-plot/blob/master/CHANGELOG.md#added).
Please see [example notebook](https://nbviewer.org/github/JetBrains/lets-plot/blob/master/docs/f-23f/geom_spoke.ipynb).

Would you check it out and see if it's missing something?

**Comment by egayer** [comment]
Hi @alshan, I hope you are doing fine too and happy new year !

Thanks for the update and thanks for adding this feature to lets-plot ! I double checked the 4.2 last week-end but I did not have time to play with it yet. `geom_spoke()` looks promising !

By the way, I meant to ask you a question but I didn't want to open an issue on this. I have been wondering if there is an easy way to get back to the default qualitative color palette and geometries size/line-width of the version 3.2 ?

Since version 4.0, the default qualitative color palette changed into [Color Brewer "Set1"] (was ["Set2"] in 3.2) and geometries default size/line-width were slightly thinner/smaller in 3.2.

You may have explained this to me a while ago but I can't remember if Let-Plot has some sort of RC file like each time Lets-Plot loads, it defines a runtime configuration containing the default styles for every plot element we create (sorry, another question from my experience with Matplotlib...). In such case it would be easy to get back to the 3.2 defaults color and line width setting :)

PS: I just want to mention that Lets-Plot became my go-to plotting package now ;)
PS2: If you are interested, I wrote a little and very simple function that generates a temporary html file and open it in your preferred web browser. I don't use PyCharm or Dataspell and rarely use notebooks. My function `ggshow_html()` allows people using the terminal, Spyder or anything else to get interactive Lets-Plots plots on their web browser (in different tabs). I have been trying to find the best/lightest web browser to do so but I couldn't figure out yet how to use Chrome for example as a desktop app (without loading bookmarks, address bar and so on) and keep the advantage of having the plots in different tabs. Let me know if offering such option within Lets-Plot is an idea that can be of any interest for you guys.

**Comment by alshan** [comment]
Hi @egayer 
> By the way, I meant to ask you a question but I didn't want to open an issue on this. I have been wondering if there is an easy way to get back to the default qualitative color palette and geometries size/line-width of the version 3.2 ?

An easy way there is no. Sizes are adjusted via parameters in `geom_xxx()` and the brewer "Set2" palette have to be set for each plot as necessary via `scale_color_brewer()`.

> Let-Plot has some sort of RC file 

No, we don't have this feature.

> In such case it would be easy to get back to the 3.2 defaults color and line width setting 

You can configure default colors via "pen", "brush" and "paper" in `theme()`, see [example](https://nbviewer.org/github/JetBrains/lets-plot/blob/master/docs/f-23c/geom_theme_colors.ipynb), but not sizes (see above). 

`ggshow_html()` is an interesting idea but I'm not getting: have you manage to get it working or its complicated?

Do you use Spyder? Did you try `plot.show()` when in Spider?

**Comment by egayer** [comment]
> An easy way there is no. Sizes are adjusted via parameters in `geom_xxx()` and the brewer "Set2" palette have to be set for each plot as necessary via `scale_color_brewer()`.

OK thanks !  I understand.

> `ggshow_html()` is an interesting idea but I'm not getting: have you manage to get it working or its complicated?
> Do you use Spyder? Did you try `plot.show()` when in Spider?

Yes I use Spyder and `plot.show()` does not show the plots. 

Before the official `plot.to_svg()` function, I made  `ggshow_svg()` that displays plots inline in the ipython console inside Spyder (and also in its plot pane). It is a very simple function that `ggsave` a plot in a temporary .svg file and then go get the temporary file to display it inside Spyder.

Now, in order to use tooltips, I made (and always use) `ggshow_html()` that again, `ggsave` a plot in a temporary .html file and then open it in a web browser. It works perfectly fine from Spyder or from any other editor like terminal or QT console.

What I meant in my previous message is that, using a web browser to display plots is not very efficient since the web browser loads bookmarks, address bar or even plugins. 

I use Chrome (but any web browser would do) and if you have 10 plots in your scripts then, the 10 plots will pop up in 10 tabs in one window. I find it very useful, but it uses more memory than actually needed. I tried, and I'd like to make this function more efficient like, asking to Chrome to not load bookmarks, address bar, plugins, etc. This can be done when asking Chrome to run in "application mode" (http://christopherroach.com/articles/jupyterlab-desktop-app/) but unfortunately, it only works for the first plot (so far), so if there are 10 plots in a script, the first one will be displayed in Chrome as application mode, but the others will be displayed in the default Chrome window, with bookmarks, address bar, plugins, etc. (lets say that I didn't dig a lot though).

Bottom line is that, `ggshow_html()` allows to pop up interactive plots from any editor (without saving them first) and it might be of interest for people using other editor than notebook, or PyCharm ;)

**Comment by alshan** [comment]
Sounds interesting ) Are you still experimenting with `ggshow_html()`?  Would be nice to have it in lets-plot.

**Comment by egayer** [comment]
It is up and running ! how shall we proceed ? should I just write the function here ?

**Comment by alshan** [comment]
Splendid, would you create a discussion an post the code there?
